internal/commands/pull/command.go

```diff
@@ -60,23 +62,47 @@ func (cmd *Command) Handler(profile *user.Profile, ui terminal.UI, clients cli.C
 		return err
 	}
 
-	pathTarget, zipPkg, err := cmd.doExport(profile, clients.Realm, app.GroupID, app.ID)
+	clientZipPkgs, err := cmd.inputs.resolveClient(ui, clients.Realm, app.GroupID, app.ID)
 	if err != nil {
 		return err
 	}
 
-	pathRelative, err := filepath.Rel(profile.WorkingDirectory, pathTarget)
+	pathTarget, zipPkg, err := cmd.doExport(profile, clients.Realm, app.GroupID, app.ID)
 	if err != nil {
 		return err
 	}
 
-	proceed, err := checkAppDestination(ui, pathTarget)
+	if clientZipPkgs != nil {
+		templatePath := filepath.Join(pathTarget, local.FrontendPath)
+		if proceed, err := checkPathDestination(ui, templatePath); err != nil {
+			return err
+		} else if !proceed {
+			return nil
+		}
+
+		for templateID, templateZipPkg := range clientZipPkgs {
+			if err := local.WriteZip(templatePath, templateZipPkg); err != nil {
+				return err
+			}
+			ui.Print(terminal.NewTextLog(fmt.Sprintf("Saved template %s to disk", templateID)))
+		}
+
+		pathTarget = filepath.Join(pathTarget, local.BackendPath)
+	}
+
+	// App path
+	proceed, err := checkPathDestination(ui, pathTarget)
 	if err != nil {
 		return err
 	} else if !proceed {
 		return nil
 	}
 
+	pathRelative, err := filepath.Rel(profile.WorkingDirectory, pathTarget)
```

Comment:
`pathRelative` here should probably remain as the relative "app path" (and not the backend path, which it is now).

additionally, we should update the `dryRun` output here to talk about templates if they're involved.  ideally keep the same existing behavior if templates arent used, but if so, we could add on to the debug logs indicating that from the "app's relative path", backend code would live in `./backend` as well as list the various front end template paths that would have been written

Reply:
👍 on the dry run updating

`pathTarget` _is_ the path where the app is located. So if there's a template involved, we put the app in a `backend` folder

Reply:
my point being: we should be showing the user the top of the directory we've created for them with this `pathRelative`, not _just_ the backend directory.  then with the dry run updates, you can add further detail explaining what the `./backend` and`./frontend` directories would contain

let me know if you want to hop on a call or slack thread to discuss these things all at once, i think a lot of these comments will end up being related and might be helpful to tackle them all at once